cloudferrylib/os/actions/transport_instance.py

```diff
@@ -57,23 +58,30 @@ def run(self, info=None, **kwargs):
             utl.INSTANCES_TYPE: {
             }
         }
-
-        # Get next one instance
-        for instance_id, instance in info[utl.INSTANCES_TYPE].iteritems():
-            instance = self._replace_user_ids(instance)
-
-            one_instance = {
-                utl.INSTANCES_TYPE: {
-                    instance_id: instance
+        processing_vms = kwargs.get('processing_vms', [])
+        try:
+            # Get next one instance
+            for instance_id, instance in info[utl.INSTANCES_TYPE].iteritems():
+                instance = self._replace_user_ids(instance)
+
+                one_instance = {
+                    utl.INSTANCES_TYPE: {
+                        instance_id: instance
+                    }
                 }
-            }
-            one_instance = self.deploy_instance(self.dst_cloud, one_instance)
-
-            new_info[utl.INSTANCES_TYPE].update(
-                one_instance[utl.INSTANCES_TYPE])
+                one_instance = self.deploy_instance(self.dst_cloud,
+                                                    one_instance)
+                dst_id = one_instance[utl.INSTANCES_TYPE].keys()[0]
+                processing_vms.append(dst_id)
+                new_info[utl.INSTANCES_TYPE].update(
+                    one_instance[utl.INSTANCES_TYPE])
+        except nova_compute.FailAfterCreateInstance as e:
+            processing_vms.append(e.inst_id)
+            raise e
```

Comment:
As a general rule, don't reraise exception like this, since it will lose original stack trace which could come in handy for debugging purposes. Just use `raise` (e.g. without `raise e`)